src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -156,22 +156,29 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
 
             // Create a token that will fire if we are disposed or if we get paused.
             using var source = CancellationTokenSource.CreateLinkedTokenSource(uncancelledToken.Value, disposalToken);
+            try
+            {
+                // Now actually synchronize the workspace.
+                var cancellationToken = source.Token;
+                cancellationToken.ThrowIfCancellationRequested();
 
-            // Now actually synchronize the workspace.
-            var cancellationToken = source.Token;
-            cancellationToken.ThrowIfCancellationRequested();
-
-            var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
-            if (client == null)
-                return;
+                var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
+                if (client == null)
+                    return;
 
-            using (Logger.LogBlock(FunctionId.SolutionChecksumUpdater_SynchronizePrimaryWorkspace, cancellationToken))
+                using (Logger.LogBlock(FunctionId.SolutionChecksumUpdater_SynchronizePrimaryWorkspace, cancellationToken))
+                {
+                    var workspaceVersion = solution.WorkspaceVersion;
+                    await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
+                        solution,
+                        (service, solution, cancellationToken) => service.SynchronizePrimaryWorkspaceAsync(solution, workspaceVersion, cancellationToken),
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException)
             {
-                var workspaceVersion = solution.WorkspaceVersion;
-                await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
-                    solution,
-                    (service, solution, cancellationToken) => service.SynchronizePrimaryWorkspaceAsync(solution, workspaceVersion, cancellationToken),
-                    cancellationToken).ConfigureAwait(false);
+                // Don't bubble up cancellation to the queue.  Just because we decided to cancel this batch isn't
+                // something the queue should be aware of.
             }
```

Comment:
note: we could add a `disposalToken.ThrowIfCancellationRequested()` here.  @sharwell  for preference.

Reply:
I wouldn't bother putting it here.